tm_cm.js

```diff
@@ -1872,3 +1872,51 @@ if (projectUrl && projectUrl != "") {
 } else {
   resetProject(true);
 }
+
+var elem = document.getElementById('cmcanvas'),
+    elemLeft = elem.offsetLeft + elem.clientLeft,
+    elemTop = elem.offsetTop + elem.clientTop,
+    context = elem.getContext('2d')
+
+var layerToDrag
+var dragOffsetX
+var dragOffsetY
+
+// Add event listener for `drag` events.
+elem.addEventListener("mousedown", dragStart, false);
+elem.addEventListener("mouseup", dragEnd, false);
+elem.addEventListener("mousemove", drag, false);
```

Comment:
Since there is only one canvas, and these events never vary, perhaps we should define them in the html rather than in the js. Undecided; this is a question of style.